### PR TITLE
feature: Pass the value of the expression through parameters

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #https://github.com/milvus-io/milvus/issues/36672

### CODE REVIEW

File: proto/milvus.proto [hunk 1]
```diff
@@ -905,6 +905,7 @@ message DeleteRequest {
   string expr = 5;
   repeated uint32 hash_keys = 6;
   common.ConsistencyLevel consistency_level = 7;
+  repeated schema.FieldData expression_values = 8;
```
Comment (liliu-z) [inline proto/milvus.proto:908]: Plz CMIIW, in #37033 we use the following code to read the value out of this `FieldData`
```
result[data.GetFieldName()] = rv[0]
```
And use ```value, ok := data[expr.GetPlaceholderName()]``` to fill the expr. And `result` above equals to `data` here.

So our assumption is that the placeholder name is the field name?

Also, can you provide some examples showing how we will fill this `expression_values`?

Comment (zhengbuqian) [inline proto/milvus.proto:908]: I wonder is `schema.FieldData` the best container to use? `schema.PlaceholderValue` seems better.

Comment (zhengbuqian) [reply proto/milvus.proto:908]: I think here `FieldData` is only used as a container to hold the template variable `name: value` mapping.

Comment (zhengbuqian) [reply proto/milvus.proto:908]: when using `schema.FieldData` we assigned new meaning to `FieldData::field_name`

**Comment by zhengbuqian** [comment]
chatted with @czs007 offline:

* the main reason we are using `schema.FieldData` is to utilize the [Base 128 variant](https://protobuf.dev/programming-guides/encoding/#varints) encoding capability to have some levels of compression on int fields. 
    * this encoding doesn't apply to floating types
    * it also doesn't apply to string types
    * Technically Base 128 variant is NOT a compression algorithm. Protobuf does not have built-in compression.
* using `fieldName` member of `schema.FieldData` as the template variable name is indeed confusing. And it will be painful if we want to modify this due to the need of backward compatibility. 

maybe the best option is to define a new proto message similar to `FieldData` but with correct member naming.

**Comment by zhengbuqian** [comment]
defining a new proto message also reduces complexity in impl. we need only a single value whereas `schema.FieldData` is a repeated, currently in [convert_field_data_to_generic_value.go](https://github.com/milvus-io/milvus/pull/37058/files#diff-6340d21c91a9b045c68a502c6fc094a02113064a51eee26985509a1814e08650) we consistently loop through the repeated field and then assert its size to be 1. In the new proto we can directly define it as not repeated.

**Pull request merged**